pkg/servicemodel/rc/v1/utils.go

```diff
@@ -226,3 +235,172 @@ func createRANParametersReportStyle2List() ([]*e2smrcies.ReportRanparameterItem,
 	reportParametersStyle2List = append(reportParametersStyle2List, ranParameter11)
 	return reportParametersStyle2List, nil
 }
+
+func (c *Client) getCellPCI(ctx context.Context, ncgi ransimtypes.NCGI) (int32, error) {
+	cell, err := c.ServiceModel.CellStore.Get(ctx, ncgi)
+	if err != nil {
+		return 0, err
+	}
+
+	return int32(cell.PCI), nil
+}
+
+func (c *Client) getARFCN(ctx context.Context, ncgi ransimtypes.NCGI) (int32, error) {
+	cell, err := c.ServiceModel.CellStore.Get(ctx, ncgi)
+	if err != nil {
+		return 0, err
+	}
+
+	return int32(cell.Earfcn), nil
+}
+
+func (c *Client) createRICIndicationFormat3(ctx context.Context, cells []ransimtypes.NCGI, subscription *subutils.Subscription, e2NodeInfoChangeID int32) (*e2appducontents.Ricindication, error) {
+	headerFormat1 := format1.NewIndicationHeader(format1.WithEventConditionID(e2NodeInfoChangeID))
+	indicationHeaderAsn1Bytes, err := headerFormat1.ToAsn1Bytes()
+	if err != nil {
+		return nil, err
+	}
+	plmnID := c.getPlmnID()
+
+	cellInfoList := make([]*e2smrcies.E2SmRcIndicationMessageFormat3Item, 0)
+	for _, ncgi := range cells {
+		cell, err := c.ServiceModel.CellStore.Get(ctx, ncgi)
+		if err != nil {
+			return nil, err
+		}
+
+		nci := ransimtypes.GetNCI(ncgi)
+		nrCGI, err := pdubuilder.CreateNrCgi(plmnID.Value().ToBytes(), &asn1.BitString{
+			Value: utils.Uint64ToBitString(uint64(nci), 36),
+			Len:   36,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		cgi, err := pdubuilder.CreateCgiNRCgi(nrCGI)
+		if err != nil {
+			return nil, err
+		}
+
+		pci, err := c.getCellPCI(ctx, ncgi)
+		if err != nil {
+			return nil, err
+		}
+		cellPCI, err := pdubuilder.CreateServingCellPciNR(pci)
+		if err != nil {
+			return nil, err
+		}
+
+		earfcn, err := c.getARFCN(ctx, ncgi)
+		if err != nil {
+			return nil, err
+		}
+		cellArfcn, err := pdubuilder.CreateServingCellArfcnNR(earfcn)
+		if err != nil {
+			return nil, err
+		}
+
+		neighborCellList := make([]*e2smrcies.NeighborCellItem, 0)
+		for _, neighborNCGI := range cell.Neighbors {
+			neighborNci := ransimtypes.GetNCI(neighborNCGI)
+			neighborNrCGI, err := pdubuilder.CreateNrCgi(plmnID.Value().ToBytes(), &asn1.BitString{
+				Value: utils.Uint64ToBitString(uint64(neighborNci), 36),
+				Len:   36,
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			neighborPci, err := c.getCellPCI(ctx, neighborNCGI)
+			if err != nil {
+				return nil, err
+			}
+
+			neighborEarfcn, err := c.getARFCN(ctx, neighborNCGI)
+			if err != nil {
+				return nil, err
+			}
+
+			neighborNrArfcn, err := pdubuilder.CreateNrArfcn(neighborEarfcn)
+			if err != nil {
+				return nil, err
+			}
+
+			nrFrequencyBandList := &e2smcommonies.NrfrequencyBandList{
```

Comment:
It looks good. BTW, since SUL information is optional and supported SUL band list can be empty list (ts 38.473 9.3.1.17), we can use it as is or update it accordingly.